keep/contextmanager/contextmanager.py

```diff
@@ -61,7 +63,11 @@ def get_full_context(self):
         full_context = {
             "providers": self.providers_context,
             "steps": self._steps_context,
-            "foreach": {"value": self.foreach_context},
+            # This is a hack to support both "before condition" and "after condition"
```

Comment:
don't understand this one at all

Reply:
added a lot of docs to the function but it still should be refactored